inappmessaging/src/main/java/com/rakuten/tech/mobile/inappmessaging/runtime/data/repositories/LocalDisplayedMessageRepository.kt

```diff
@@ -134,18 +134,23 @@ internal interface LocalDisplayedMessageRepository {
                     (onLaunch || user != AccountRepository.instance().userInfoHash)) {
                 user = AccountRepository.instance().userInfoHash
                 // reset message list from cached using updated user info
-                val sharedPref = InAppMessaging.instance().getSharedPref()
-
-                resetDisplayed(sharedPref)
+                resetDisplayed()
             }
         }
 
-        @SuppressWarnings("TooGenericExceptionCaught")
-        private fun resetDisplayed(sharedPref: SharedPreferences?) {
+        @SuppressWarnings("TooGenericExceptionCaught", "LongMethod")
+        private fun resetDisplayed() {
             val listString = try {
-                sharedPref?.getString(LOCAL_DISPLAYED_KEY, "") ?: ""
+                InAppMessaging.instance().getHostAppContext()?.let { it ->
+                    PreferencesUtil.getString(
+                        it,
+                        "internal_shared_prefs_" + AccountRepository.instance().userInfoHash,
```

Comment:
We can update SharedPreferencesUtil to have this feature. This will introduce a lot of duplicate codes if we always have the "internal_shared_prefs_" string.

Reply:
updated